File: frontend/src/lib/types/piscicole.ts
// lib/types/piscicole.ts
import type { AnimalBase } from './animal';

export interface PiscicoleBase extends AnimalBase {
    production_viande: boolean;
    production_reproduction: boolean;
    taille_moyenne?: number;
    poids_moyen?: number;
}

export interface PiscicoleCreate extends PiscicoleBase { }
export interface PiscicoleUpdate extends Partial<PiscicoleBase> { }

export interface PiscicoleResponse extends PiscicoleBase {
    id: number;
    created_at: string;
    updated_at: string;
}

export interface PiscicoleStats {
    total_piscicoles: number;
    production_viande: number;
    production_reproduction: number;
    taille_moyenne_globale: number;
    taille_min: number;
    taille_max: number;
}

File: backend/app/schemas/piscicole.py
# backend/app/schemas/piscicole.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from .animal import AnimalBase


class PiscicoleBase(AnimalBase):
    production_viande: bool = False
    production_reproduction: bool = False
    taille_moyenne: Optional[float] = Field(None, ge=0)


class PiscicoleCreate(PiscicoleBase):
    pass


class PiscicoleUpdate(BaseModel):
    production_viande: Optional[bool] = None
    production_reproduction: Optional[bool] = None
    taille_moyenne: Optional[float] = Field(None, ge=0)


class PiscicoleResponse(PiscicoleBase):
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True